Dockerfile

```diff
@@ -1,22 +1,32 @@
-FROM golang:1.11.4-alpine3.8 AS builder
-LABEL maintainer="joona@kuori.org"
+FROM golang:1.12.0-alpine3.9 AS builder
+
+ENV ACME_DNS_VERSION=v0.7.2
+
+WORKDIR /go/src/github.com/joohoi
 
-RUN apk add --update gcc musl-dev git
+RUN apk add --no-cache gcc musl-dev git && \
+    git clone -b ${ACME_DNS_VERSION} --depth 1 https://github.com/joohoi/acme-dns.git acme-dns
 
-RUN go get github.com/joohoi/acme-dns
 WORKDIR /go/src/github.com/joohoi/acme-dns
-RUN CGO_ENABLED=1 go build
 
-FROM alpine:latest
+RUN CGO_ENABLED=1 go build -ldflags="-s -w -linkmode external -extldflags -static"
 
-WORKDIR /root/
-COPY --from=builder /go/src/github.com/joohoi/acme-dns .
-RUN mkdir -p /etc/acme-dns
-RUN mkdir -p /var/lib/acme-dns
-RUN rm -rf ./config.cfg
-RUN apk --no-cache add ca-certificates && update-ca-certificates
+RUN mkdir -p /build/etc/acme-dns && \
+    mkdir -p /build/var/lib/acme-dns && \
+    cp /go/src/github.com/joohoi/acme-dns/acme-dns /build && \
+    cp /go/src/github.com/joohoi/acme-dns/config.cfg /build/etc/acme-dns && \
+    sed -Ei -e "s/(port ?= ?)\"80\"/\1\"8080\"/" \
+            -e "s/(listen ?= ?)\".*:53\"/\1\":5353\"/" /build/etc/acme-dns/config.cfg && \
+    chown 1000:0 /build/etc/acme-dns /build/var/lib/acme-dns && \
+    chmod ug+w /build/etc/acme-dns /build/var/lib/acme-dns && \
+    chmod +x /build/acme-dns
 
+
+FROM scratch
```

Comment:
I really hate this trend. How do you plan to do health checking with this image?

Reply:
`GET /health`?

Reply:
There is no GET command in `scratch`

Here is a potential solution.
https://github.com/Soluto/golang-docker-healthcheck-example

Reply:
Ah, I see. Didn't realize health checks were something Docker had built-in. Too bad the HEALTHCHECK instruction only allows checking by running commands, and not via HTTP queries.

Reply:
There doesn't seem to be a lot of desire to support HTTP health checking.

https://github.com/moby/moby/pull/25364#issuecomment-276116088

Reply:
Why? It's just a binary, that does not have any dependencies to a "full" linux environment. I see your point with the healthcheck, that this may make sense to switch to an alpine and add curl for the checks.
But why not just add another go-binary that does the healthcheck? Like in the link you commented.

Reply:
I used the `FROM scratch` approach multiple times for different go binaries without any troubles, just the debugging was not that easy as if I used alpine as a base image.
I will change it to alpine, as this may make the debugging of problems a bit easier.